### PR TITLE
Edition 2018

### PR DESCRIPTION
**Pull request by fkohlgrueber**
Hi, 

this PR updates `term` to edition 2018. I've also used `clippy` and `rustfmt` to fix smaller issues.

Please let me know if there are issues within the changes I made. Thanks in advance ;-)

### CODE REVIEW

File: src/terminfo/mod.rs [hunk 1]
```diff
@@ -114,11 +113,11 @@ impl TermInfo {
             Ok(TermInfo {
                 names: vec![name.to_owned()],
                 bools: HashMap::new(),
-                numbers: numbers,
-                strings: strings,
+                numbers,
+                strings,
```
Comment (Stebalien) [inline src/terminfo/mod.rs]: Let's not do this: https://github.com/rust-lang/rust-clippy/issues/3422. I'd rather just disable the lint.

File: src/terminfo/mod.rs [hunk 2]
```diff
@@ -372,7 +374,7 @@ impl<T: Write> TerminfoTerminal<T> {
         };
 
         TerminfoTerminal {
-            out: out,
+            out,
```
Comment (Stebalien) [inline src/terminfo/mod.rs]: Same.

File: src/terminfo/parm.rs [hunk 3]
```diff
@@ -115,7 +115,7 @@ pub struct Variables {
     /// Static variables A-Z
     sta: [Param; 26],
     /// Dynamic variables a-z
-    dyn: [Param; 26],
+    r#dyn: [Param; 26],
```
Comment (Stebalien) [inline src/terminfo/parm.rs]: Let's just rename these, they're private anyways.

**Comment by fkohlgrueber** [comment]
Ok, I've updated the PR. I renamed `#dyn` to `dyn_vars` and `sta` to `sta_vars` for consistency. Hope that's ok for you.

**Pull request merged**